app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -406,7 +406,7 @@ def look_up_host(hostname, ipaddr, opts = {})
     # we looked-up does not have a different ems_ref and is not
     # owned by another provider, this would cause us to overwrite
     # a different host record
-    if (opts[:ems_ref] && h.ems_ref != opts[:ems_ref]) || (opts[:ems_id] && h.ems_id != opts[:ems_id])
+    if (opts[:ems_ref] && h.ems_ref != opts[:ems_ref]) || (opts[:ems_id] && h.ems_id && h.ems_id != opts[:ems_id])
```

Comment:
Not checking for `ems_id.present?` was causing archived hosts to be skipped because `nil != ems_id` was always true